src/http-adapter.js

```diff
@@ -109,39 +123,29 @@ class HttpAdapter {
     }
     promise.then(removeFromOutstandingRequests, removeFromOutstandingRequests)
 
-    return promise
+    return await promise
   }
 
-  _makeRequest ({ url, type, headers, data }) {
-    return new Promise((resolve, reject) => {
-      let request = {
-        url,
-        type,
-        headers,
-        success: (data, textStatus, jqXhr) => {
-          if (!data && jqXhr.status !== 204) {
-            throw new Error(`request returned ${jqXhr.status} status without data`)
-          }
-          return resolve(data)
-        },
-        error: (response) => {
-          if (response.readyState === 0 || response.status === 0) {
-            // this is a canceled request, so we literally should do nothing
-            return
-          }
-
-          const error = new Error(`request for resource, ${url}, returned ${response.status} ${response.statusText}`)
-          error.response = response
-          reject(error)
-        },
-        // being explicit about data type so jQuery doesn't "intelligent guess" wrong
-        // changing this may not break tests, but does behave badly in prod
-        dataType: 'text',
-        data
-      }
-
-      ajax(request)
+  async _makeRequest ({method, url, headers, body}) {
+    // eslint-disable-next-line no-undef
+    let response = await fetch(url, {
+      method,
+      headers,
+      body
     })
+    if (response.status < 300 && response.status >= 200) {
+      if (response.body == null && response.status !== 204) {
```

Comment:
I don't believe this is equivalent to the previous condition because `body` is a `ReadableStream`. An empty `ReadableStream` will not satisfy `== null`, but in jQuery version it would produce `''` which would satisfy `!data`.

I tested this on Chrome with an API endpoint that returns a 204 and it did have a `ReadableStream` for `Response.body`, though it was empty.

Reply:
Can [response.ok](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok) be used here instead?

```
if (response.ok) {
 return await resonse.json()
}
```

The body does return a [ReadableStream](https://developer.mozilla.org/en-US/docs/Web/API/Body/body) 
I think you can get it with `const body = await response.text()`

You should also be able to return early when given a 204.
```
if (response.status === 204) return {}
```

Reply:
Good points. Let me try some of the suggestions. I could have sworn the body was undefined but let me double check. I should also add a test for this

Reply:
Ok so I came up with the following:

`
if (response.ok) {
      if(response.status === 204) return {}

      const respBodyStringified = await response.text()

      if (!respBodyStringified) {
        throw new Error(
          `request returned ${response.status} status without data`
        )
      }

      return JSON.parse(respBodyStringified)
    }
`

we can only consume the response once, so we either can call .text() or .json() on it. If the status is not a 204, we can see what data body has been returned. If it's falsy, throw an error. Otherwise parse it to JSON. 

I tested this logic on a 204 with the first condition removed, so it would be like testing a 200 with ` await response.text()` returning `""`